src/messaging/ExchangeContext.cpp

```diff
@@ -401,5 +473,16 @@ CHIP_ERROR ExchangeContext::HandleMessage(const PacketHeader & packetHeader, con
     return err;
 }
 
+CHIP_ERROR ExchangeContext::OnAttestedDevice(SecureSessionHandle session, OperationalCredentialSet * opCredSet,
```

Comment:
I don't think that `ExchangeContext` should be aware of specifics of device attestation. The message handling of specific types should be one layer above exchange context.